### ISSUE TITLE
launching exoskeleton simulation opens with immediate erratic movement

### ISSUE THREAD

**Issue by Bart-van-Ingen** [project-march/simulation#13]
when launching march_world.launch, Gazebo opens and the exoskeleton limbs are moving erratically without a command being sent. Even commenting out the controllers has no effect.

**Comment by Ishadijcks** [comment]
Can confirm, it might be related to it starting outside of its limits, but I am not familiar enough with Gazebo to conclude anything yet.

**Comment by Ishadijcks** [comment]
@Bart-van-Ingen can you give this a try?
http://answers.gazebosim.org/question/8994/jackal-simulator-flipping-and-flying-in-gazebo-51/

**Comment by Ishadijcks** [comment]
https://answers.ros.org/question/216092/problem-with-the-robot-model-collapsing/?answer=216098%3Fanswer%3D216098#post-id-216098

**Comment by Bart-van-Ingen** [comment]
I think I have found what is going wrong. the change in soft limits seems to be causing the problem. Originally these were set to 5 and are now set to 20000 in the xacro. This was caused by the recently merged torque mode feature, in which these limits were changed to facilitate the IMC. This did not take into account that even though the simulation uses the same xacro, it doesn't use the march hardware and therefore also not the relevant hardware interface. It uses the interface that gazebo provides which uses as unit the Newton meter (I think) as input for the joint. When a simulation is started, the joints have some movement and therefore speed. I believe that the largek_velocity value a disproportionate response is generated, causing the frantic behaviour. Either a way needs to be found in which this limits change depending on the launch file being launched, or two urdfs need to be used.
